--- a/util/grub.d/10_linux.in~	2012-04-18 23:24:38.000000000 +0200
+++ b/util/grub.d/10_linux.in	2012-11-28 22:27:41.652673833 +0100
@@ -29,7 +29,14 @@
 CLASS="--class gnu-linux --class gnu --class os"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
+  if [ -f "/etc/milis-release" ] ; then
+    OS="$(sed 's, release .*$,,g' /etc/milis-release)"
+  elif [ -e "/etc/system-release" ] ; then
+    OS="$(sed 's, release .*$,,g' /etc/system-release)"
+  else
+    OS=GNU/Linux
+  fi
 else
   OS="${GRUB_DISTRIBUTOR} GNU/Linux"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
