--- blackbox-0.70.1/lib/Image.cc.orig	2008-10-03 09:23:56.000000000 +0200
+++ blackbox-0.70.1/lib/Image.cc	2008-10-03 09:24:08.000000000 +0200
@@ -44,6 +44,7 @@
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 // #define COLORTABLE_DEBUG
 // #define MITSHM_DEBUG
--- blackbox-0.70.1.orig/lib/Resource.cc	2008-10-07 18:32:55.000000000 +0200
+++ blackbox-0.70.1/lib/Resource.cc	2008-10-07 18:34:05.000000000 +0200
@@ -29,6 +29,7 @@
 #include <X11/Xresource.h>
 
 #include <stdio.h>
+#include <strings.h>
 
 
 bt::Resource::Resource(void)
--- blackbox-0.70.1.orig/lib/XDG.cc	2008-10-07 18:32:55.000000000 +0200
+++ blackbox-0.70.1/lib/XDG.cc	2008-10-07 18:47:25.000000000 +0200
@@ -26,7 +26,7 @@
 #include "XDG.hh"
 
 #include <stdlib.h>
-
+#include <algorithm>
 
 // make sure directory names end with a slash
 static std::string terminateDir(const std::string &string)
--- blackbox-0.70.1.orig/src/BlackboxResource.cc	2008-10-07 18:32:55.000000000 +0200
+++ blackbox-0.70.1/src/BlackboxResource.cc	2008-10-07 18:48:13.000000000 +0200
@@ -33,6 +33,7 @@
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 
+#include <strings.h>
 
 BlackboxResource::BlackboxResource(const std::string& rc): rc_file(rc) {
   screen_resources = 0;
--- blackbox-0.70.1.orig/src/ScreenResource.cc	2008-10-07 18:32:55.000000000 +0200
+++ blackbox-0.70.1/src/ScreenResource.cc	2008-10-07 18:49:10.000000000 +0200
@@ -32,7 +32,7 @@
 #include <Resource.hh>
 
 #include <assert.h>
-
+#include <strings.h>
 
 static const int iconify_width  = 9;
 static const int iconify_height = 9;
--- blackbox-0.70.1.orig/src/Screen.cc	2008-10-07 18:32:55.000000000 +0200
+++ blackbox-0.70.1/src/Screen.cc	2008-10-07 18:50:01.000000000 +0200
@@ -47,7 +47,7 @@
 #include <assert.h>
 #include <ctype.h>
 #include <dirent.h>
-
+#include <string.h>
 
 static bool running = true;
 static int anotherWMRunning(Display *, XErrorEvent *) {
--- blackbox-0.70.1.orig/src/main.cc	2008-10-07 18:32:55.000000000 +0200
+++ blackbox-0.70.1/src/main.cc	2008-10-07 18:50:48.000000000 +0200
@@ -35,7 +35,7 @@
 #include "../version.h"
 
 #include <stdio.h>
-
+#include <string.h>
 
 static void showHelp(int exitval) {
   // print version - this should not be localized!
--- blackbox-0.70.1.orig/util/bsetroot.cc	2008-10-07 18:32:55.000000000 +0200
+++ blackbox-0.70.1/util/bsetroot.cc	2008-10-07 18:52:33.000000000 +0200
@@ -31,7 +31,8 @@
 
 #include <X11/Xatom.h>
 #include <stdio.h>
-
+#include <string.h>
+#include <stdlib.h>
 
 // ignore all X errors
 static int x11_error(::Display *, XErrorEvent *)

--- blackbox-0.70.1.orig/lib/Util.hh
+++ blackbox-0.70.1/lib/Util.hh
@@ -25,6 +25,8 @@
 #ifndef __Util_hh
 #define __Util_hh

+#include <X11/Xutil.h>
+
 #include <limits.h>
 #include <string>

@@ -94,10 +96,8 @@

   std::string tolower(const std::string &string);

-#ifdef _XUTIL_H_
   std::string textPropertyToString(::Display *display,
                                    ::XTextProperty& text_prop);
-#endif

 } // namespace bt
