--- vboxdrv/Makefile	2016-03-04 18:47:50.000000000 +0200
+++ vboxdrv.new/Makefile	2016-03-08 13:04:56.287601296 +0200
@@ -300,7 +300,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; \
 	rm -f /etc/vbox/module_not_compiled
 
 install_rpm: $(MODULE)
--- vboxnetadp/Makefile	2016-03-04 18:47:50.000000000 +0200
+++ vboxnetadp.new/Makefile	2016-03-08 13:05:15.741103737 +0200
@@ -203,7 +203,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; \
 	rm -f /etc/vbox/module_not_compiled
 
 install_rpm: $(MODULE)
--- vboxnetflt/Makefile	2016-03-04 18:47:49.000000000 +0200
+++ vboxnetflt.new/Makefile	2016-03-08 13:05:30.709490190 +0200
@@ -207,7 +207,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; \
 	rm -f /etc/vbox/module_not_compiled
 
 install_rpm: $(MODULE)
--- vboxpci/Makefile	2016-03-04 18:47:50.000000000 +0200
+++ vboxpci.new/Makefile	2016-03-08 13:06:08.318460615 +0200
@@ -205,8 +205,7 @@
 
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
-	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a;
+	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR);
 
 install_rpm: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
