--- asterisk-13.2.0/menuselect-tree	2015-04-01 13:33:21.374042590 +0900
+++ asterisk-13.2.0/menuselect-tree.sounds-en-gsm-disable	2015-04-01 13:43:09.645045100 +0900
@@ -1877,7 +1877,7 @@
 		</member>
 		<member name="CORE-SOUNDS-EN-GSM" displayname="English, GSM format" >
 			<support_level>core</support_level>
-			<defaultenabled>yes</defaultenabled>
+			<defaultenabled>no</defaultenabled>
 		</member>
 		<member name="CORE-SOUNDS-EN-G729" displayname="English, G.729 format">
 			<support_level>core</support_level>
