--- configure.ac	2007/03/03 13:44:14	1.21
+++ configure.ac	2007/05/06 00:18:32	1.22
@@ -1,11 +1,11 @@
 dnl Process this file with autoconf to produce configure.
 # vim: noet sw=8
-# $Id: configure.ac,v 1.21 2007/03/03 13:44:14 stevecheng Exp $
+# $Id: configure.ac,v 1.22 2007/05/06 00:18:32 stevecheng Exp $
 
-AC_INIT([docbook2X], 0.8.8, [stevecheng@users.sourceforge.net], docbook2X)
+AC_INIT([docbook2X], 0.8.9, [stevecheng@users.sourceforge.net], docbook2X)
 AC_CONFIG_SRCDIR(perl/docbook2X.pl)
 
-AM_INIT_AUTOMAKE(docbook2X, 0.8.8)
+AM_INIT_AUTOMAKE(docbook2X, 0.8.9)
 
 XSLT_PROCESSOR="libxslt"
 
@@ -149,6 +149,11 @@
 dnl Ensure that all static_* are fully expanded.
 
 eval static_datadir="$datadir"
+old_val=""
+until test "$static_datadir" = "$old_val"; do
+	old_val="$static_datadir"
+	eval static_datadir="$static_datadir"
+done
 
 eval static_bindir="$bindir"
 old_val=""